src/PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.cpp

```diff
@@ -209,6 +215,222 @@ Scalar<DataType> time_deriv_of_lapse(
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi) noexcept {
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    for (size_t j = 0; j < SpatialDim; ++j) {
+      for (size_t a = 0; a < SpatialDim + 1; ++a) {
+        for (size_t b = 0; b < SpatialDim + 1; ++b) {
+          deriv_shift->get(i,j) += (inverse_spacetime_metric.get(j+1, a)
+                + spacetime_unit_normal.get(j+1) * spacetime_unit_normal.get(a))
+            * spacetime_unit_normal.get(b) * phi.get(i, a, b);
+        }
+      }
+      deriv_shift->get(i,j) *= get(lapse);
+    }
+  }
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::iJ<DataType, SpatialDim, Frame> spatial_deriv_of_shift(
+    const Scalar<DataType>& lapse,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi) noexcept {
+  auto deriv_shift =
+    make_with_value<tnsr::iJ<DataType, SpatialDim, Frame>>(get(lapse), 0.);
+  GeneralizedHarmonic::spatial_deriv_of_shift<SpatialDim, Frame, DataType>(
+    make_not_null(&deriv_shift), lapse, inverse_spacetime_metric,
+    spacetime_unit_normal, phi);
+  return deriv_shift;
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void time_deriv_of_shift(
+    const gsl::not_null<tnsr::I<DataType, SpatialDim, Frame>*> dt_shift,
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi) noexcept {
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    for (size_t j = 0; j < SpatialDim; ++j) {
+      for (size_t a = 0; a < SpatialDim + 1; ++a) {
+        // first term
+        dt_shift->get(i) -= get(lapse) * pi.get(j+1, a)
+            * spacetime_unit_normal.get(a) * inverse_spatial_metric.get(i, j);
+        // second term
+        for (size_t k = 0; k < SpatialDim; ++k) {
+          dt_shift->get(i) += shift.get(j)
+              * spacetime_unit_normal.get(a) * phi.get(j, k+1, a)
+              * inverse_spatial_metric.get(i, k);
+        }
+      }
+    }
+    dt_shift->get(i) *=  get(lapse);
+  }
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::I<DataType, SpatialDim, Frame> time_deriv_of_shift(
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi) noexcept {
+  auto dt_shift =
```

Comment:
default construct

Reply:
#413 says 
>  The solution is to use make_with_value when constructing Tensor, or specifying the number of grid points at construction.

I guess we're doing the latter here.